libqfieldsync/project_checker.py

```diff
@@ -439,3 +444,47 @@ def check_external_layers(
             )
 
         return None
+
+    def check_layer_has_experimental_cloud_support(
+        self, layer_source: LayerSource
+    ) -> Optional[FeedbackResult]:
+        """Check if layer has experimental cloud support"""
+        layer = layer_source.layer
+
+        # Check if layer is marked for offline editing
+        if layer_source.cloud_action != SyncAction.OFFLINE:
+            return None
+
+        # Check if layer is read-only
+        if layer.readOnly():
+            return None
+
+        # Check if layer is in a supported format
+        provider_type = layer.providerType()
```

Comment:
```suggestion
	if not layer.dataProvider():
	    return None

        provider_type = layer.providerType()
        storage_type = layer.dataProvider().storageType()
```

You can get and cache this information.